ansible_collections/arista/avd/plugins/module_utils/eos_designs.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from functools import cached_property
+try:
+    from ansible_collections.arista.avd.plugins.filter.convert_dicts import convert_dicts
+    from ansible_collections.arista.avd.plugins.filter.list_compress import list_compress
+    from ansible_collections.arista.avd.plugins.filter.range_expand import range_expand
+    from ansible_collections.arista.avd.plugins.filter.natural_sort import natural_sort
+except ImportError as imp_exc:
+    FILTER_IMPORT_ERROR = imp_exc
+else:
+    FILTER_IMPORT_ERROR = None
+
+
+class AVDFacts:
+
+    def __init__(
+        self,
+        hostvars=None,
+        combine=None,
+        template_lookup_module=None
+    ):
+        if hostvars:
+            self._hostvars = hostvars
+        if combine:
+            self._combine = combine
+        if template_lookup_module:
+            self._template_lookup_module = template_lookup_module
```

Comment:
What is the rational behind not setting the instance attributes if they are `None`? Not saying this is wrong, just trying to understand why we would rather **not** set an attribute than havig it being `None`

Reply:
That is to avoid overwriting variables set by a previous template in the template stack. (In this case "template" == "python class"). Since we run a merge on the data, we don't want to get a `nxt` with `None`, since that will blank out any data we already have.
Only exception is the custom_structured_config stuff, where we allow the user to null any variable.

Reply:
I missed the area where we are doing this. You are correct, we can just set to the passed value. Actually we should probably remove the default value since the class will not work without these set correctly.